extensions/wikia/Privacy/RemoveUserDataOnWikiTask.php

```diff
@@ -0,0 +1,130 @@
+	use Loggable;
+
+	/**
+	 * Deletes all CheckUser records associated with the given user
+	 *
+	 * @param int $userId
+	 * @throws DBUnexpectedError
+	 * @return true if operation was successful
+	 */
+	private function removeCheckUserData( int $userId ) {
+		try {
+			$db = wfGetDB( DB_MASTER );
+			// remove check user data
+			$db->delete( 'cu_changes', ['cuc_user' => $userId], __METHOD__ );
+			$db->delete( 'cu_log', ['cul_target_id' => $userId], __METHOD__ );
+			$this->info( "Removed CheckUser data", ['user_id' => $userId] );
+			return true;
+		} catch( DBError $error ) {
+			$this->error( "Couldn't remove CheckUser data", ['exception' => $error, 'user_id' => $userId] );
+			return false;
+		}
+	}
+
+	/**
+	 * Removes the IP address from all RecentChanges records associated with the given user
+	 *
+	 * @param int $userId
+	 * @return true if operation was successful
+	 */
+	private function removeIpFromRecentChanges( int $userId ) {
+		try {
+			$db = wfGetDB( DB_MASTER );
+			$db->update( 'recentchanges', ['rc_ip_bin' => ''], ['rc_user' => $userId], __METHOD__ );
+			$this->info( "Removed IPs from recent changes", ['user_id' => $userId] );
+			return true;
+		} catch( DBError $error ) {
+			$this->error( "Couldn't remove IP from recent changes", ['exception' => $error, 'user_id' => $userId] );
+			return false;
+		}
+	}
+
+	/**
+	 * Removes abuse filter logs associated with the given user
+	 *
+	 * @param int $userId
+	 * @return true if operation was successful
+	 */
+	private function removeAbuseFilterData( int $userId ) {
+		global $wgEnableAbuseFilterExtension;
+		if( $wgEnableAbuseFilterExtension ) {
+			try {
+				$db = wfGetDB( DB_MASTER );
+				$db->update( 'abuse_filter', ['af_user_text' => ''], ['af_user' => $userId], __METHOD__ );
+				$db->update( 'abuse_filter_history', ['afh_user_text' => ''], ['afh_user' => $userId], __METHOD__ );
+				$db->delete( 'abuse_filter_log', ['afl_user' => $userId], __METHOD__ );
+				$this->info( "Removed abuse filter data", ['user_id' => $userId] );
+				return true;
+			} catch ( DBError $error) {
+				$this->error( "Couldn't remove abuse filter data", ['exception' => $error, 'user_id' => $userId] );
+				return false;
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Permanently removes the userpage and user's talk, blog and wall pages.
+	 *
+	 * Since this method relies on the username, it may be impossible to retry this operation
+	 * after the user's global data is removed.
+	 *
+	 * @param string $username
+	 * @return true if operation was successful
+	 */
+	private function removeUserPages( string $username ) {
+		global $wgEnableBlogArticles, $wgEnableWallExt;
+		try {
+			$namespaces = [NS_USER, NS_USER_TALK];
+			if( $wgEnableBlogArticles ) {
+				$namespaces[] = NS_BLOG_ARTICLE;
+			}
+			if( $wgEnableWallExt ) {
+				$namespaces[] = NS_USER_WALL_MESSAGE_GREETING;
+				$namespaces[] = NS_USER_WALL_MESSAGE;
+				$namespaces[] = NS_USER_WALL;
```

Comment:
let's remove the wg conditions and use literals instead of constants so we cover also wikis where wall or blogs used to be enabled then got turned off